Framework/Algorithms/src/DetectorEfficiencyCor.cpp

```diff
@@ -317,8 +317,8 @@ void DetectorEfficiencyCor::getDetectorGeometry(
       detAxis = V3D(0, 1, 0);
       // assume radi in z and x and the axis is in the y
       PARALLEL_CRITICAL(deteff_shapecachea) {
-        m_shapeCache.insert(std::pair<const Object *, std::pair<double, V3D>>(
-            shape_sptr.get(), std::pair<double, V3D>(detRadius, detAxis)));
+        m_shapeCache.emplace(shape_sptr.get(),
+                             std::make_pair(detRadius, detAxis));
```

Comment:
Just checked the `emplace` documentation: It supports a variable argument list just like `emplace_back`, i.e., this could be written as

```cpp
m_shapeCache.emplace(shape_sptr.get(), detRadius, detAxis);
```